pkg/registry/apis/provisioning/safepath/limit.go

```diff
@@ -0,0 +1,29 @@
+package safepath
+
+import (
+	"errors"
+	"strings"
+)
+
+var (
+	ErrPathTooLong = errors.New("path too long")
+	ErrPathTooDeep = errors.New("path too deep")
+)
+
+const (
+	MaxPathLength = 1024 // Maximum allowed path length in characters
+	MaxNestDepth  = 8    // Maximum allowed directory nesting depth
+)
+
+func ValidatePath(path string) error {
+	if len(path) > MaxPathLength {
+		return ErrPathTooLong
+	}
+
+	segments := strings.Split(path, "/")
+	if len(segments) > MaxNestDepth {
```

Comment:
```suggestion
	if strings.Split(path, "/") > MaxNestDepth {
```
since we don't use the value, can we just count how many exist?   (we could either add one or change the Depth constant since root folder would have count=0, but segments =1 etc